### PR TITLE
adding the first Oracle Cloud check 

### PR DESCRIPTION
**Pull request by louwersj**
Adding the first Oracle Cloud check, it will check on the use of an IP from the external IP pool versus an IP from the internal IP pool. Warning for the use of publicly routable IPs which could expose a customer service to the public internet.

### CODE REVIEW

File: internal/app/tfsec/checks/oci001.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				"https://registry.terraform.io/providers/hashicorp/opc/latest/docs/resources/opc_compute_ip_address_reservation",
+				"https://registry.terraform.io/providers/hashicorp/opc/latest/docs/resources/opc_compute_instance",
+			},
+		},
+		Provider:       scanner.OracleProvider,
+		RequiredTypes:  []string{"resource"},
+		RequiredLabels: []string{"opc_compute_ip_address_reservation"},
+		CheckFunc: func(check *scanner.Check, block *parser.Block, _ *scanner.Context) []scanner.Result {
+			if attr := block.GetAttribute("ip_address_pool"); attr != nil {
+				if attr.IsAny("public-ippool") {
```
Comment (owenrumney) [inline internal/app/tfsec/checks/oci001.go:52]: If there is only one value with is an issue you can use `attr.Equals("public-ippool", parser.IngnoreCase)`

**Review by owenrumney — COMMENTED** [review]
Looks good thanks, could you add tests like the other checks too?

File: internal/app/tfsec/checks/oci001.go (hunk 1 above)
Comment (louwersj) [reply internal/app/tfsec/checks/oci001.go:52]: Found the testing pieces... I will start writing one... 

Please do have some patience with me, it is a learning curve which we need to go through. As soon as we have nailed this I think we can get some people within my team who can leverage the learning curve I am taking right ow and help adding new checks ;)

Comment (owenrumney) [reply internal/app/tfsec/checks/oci001.go:52]: Hey @louwersj no pressure at all, really appreciate you helping out

Comment (louwersj) [reply internal/app/tfsec/checks/oci001.go:52]: Added the first attempt to write a test for the OCI001 check...

**Review by louwersj — COMMENTED** [review]
Test added.

File: internal/app/tfsec/checks/oci001.go (hunk 1 above)
Comment (owenrumney) [reply internal/app/tfsec/checks/oci001.go:52]: Hey @louwersj  - looking at the docs, it seems like `ip_address_pool` attribute is better as an `Equals`

**Comment by louwersj** [comment]
@owenrumney can you please have a look at the inclusion of the test? I think with adding the test the first OCI check (and test) should be OK for a merge. Currently the merge is awaiting your reviewers approval.

**Review by owenrumney — APPROVED** [review]
looks great, thanks

**Pull request merged**